pkg/resources/validation/namespace/namespace.go

```diff
@@ -38,17 +39,49 @@ func (v *Validator) GVR() schema.GroupVersionResource {
 }
 
 // Operations returns list of operations handled by this validator.
-func (v *Validator) Operations() []admissionregistrationv1.OperationType {
-	return []admissionregistrationv1.OperationType{
-		admissionregistrationv1.Update,
-		admissionregistrationv1.Create,
+func (v *Validator) Operations() []admv1.OperationType {
+	return []admv1.OperationType{
+		admv1.Update,
+		admv1.Create,
 	}
 }
 
 // ValidatingWebhook returns the ValidatingWebhook used for this CRD.
-func (v *Validator) ValidatingWebhook(clientConfig admissionregistrationv1.WebhookClientConfig) *admissionregistrationv1.ValidatingWebhook {
+func (v *Validator) ValidatingWebhook(clientConfig admv1.WebhookClientConfig) []admv1.ValidatingWebhook {
 	// Note that namespaces are actually CLUSTER scoped
-	return admission.NewDefaultValidatingWebhook(v, clientConfig, admissionregistrationv1.ClusterScope)
+
+	// standardWebhook validates all operations other than create on all namespaces.
+	standardWebhook := admission.NewDefaultValidatingWebhook(v, clientConfig, admv1.ClusterScope, []admv1.OperationType{admv1.Update})
```

Comment:
The comment says "all operations other than create", but the operation type is only `Update`. Is that intentional? Don't we need to also add `Delete` here? Do we need to add a test case to cover this?

https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/api/admissionregistration/v1/types.go#L536-L542

Reply:
I can update the comment. I meant all operations the handler cares about, not all available operations. This handler does not care about the deletion of a namespace since its goal is to check the PSA labels on active namespaces. 

We don't need a test case for deletion since this handler will never be sent a deletion request. This is controlled by another package that filters requests based on the specified options in this function https://github.com/rancher/webhook/blob/release/v0.3/pkg/resources/validation/namespace/namespace.go#L41-L44